### ISSUE TITLE
Error using Mininet in Vagrant option.

### ISSUE THREAD

**Issue by GonzaloFajardoMedina** [fp7-netide/IDE#18]
Hi, 
when I run Mininet, shows the next error: 

![error](https://cloud.githubusercontent.com/assets/11718117/16043395/bebe112c-323f-11e6-835d-225adc456482.png)

Any idea?

Thanks everybody.

**Comment by ElisaRojas** [comment]
Just to clarify a bit more, the problem that @GonzaloFajardoMedina has is that Mininet is not able to connect to the Engine. 

Maybe some of the modules of the Engine are not running?

**Comment by GonzaloFajardoMedina** [comment]
No, all modules are running: Core, Shim and ryu-backend. And the shim and the ryu-backend connect to the Core.

**Comment by cstritzke** [comment]
It's all a matter of time when it comes to OpenDaylight. If you wait for about half a minute, Mininet shouldn't print that error anymore.
But, error or not, it should work once ODL is fully initialized.

**Comment by GonzaloFajardoMedina** [comment]
Although I wait that ODL initializes (about 2 minutes after it is initialized), Mininnet still does not go. I also tested the ryu-shim option and it still gives the error: **unable to contact the remote controller at 127.0.0.1:6633** .
any suggestion?

**Comment by cstritzke** [comment]
Try to set the controller port in your model to 6644, then regenerate it.

**Comment by GonzaloFajardoMedina** [comment]
Hi @cstritzke,
I still get the same error. What could I do?
I remember recently changed the Core.
Thanks.

**Comment by ElisaRojas** [comment]
Hi @cstritzke ,

I have the same problem that @GonzaloFajardoMedina when using Ryu+Core+Ryu. I've tried stopping and restarting the backend, and also Mininet, but yet the same error:

![mininet](https://cloud.githubusercontent.com/assets/2779496/18164903/b3c87fd2-7042-11e6-8905-84430c19b365.png)

It's a bit strange, because Ryu should run quite fast, it shouldn't need extra time to be ready. I waited and tried again just in case with no success.

I'm attaching below screenshots from the core, shim and backend just in case.

![core](https://cloud.githubusercontent.com/assets/2779496/18164891/a709f35c-7042-11e6-9805-eb9c449b83cc.png)

![shim](https://cloud.githubusercontent.com/assets/2779496/18164893/a9f24718-7042-11e6-9f4e-34ff19da9ff9.png)

![backend](https://cloud.githubusercontent.com/assets/2779496/18164897/ae48e8e4-7042-11e6-834d-330f78772020.png)

I've also checked the Ryu starters and they look good.
Any idea of what could it be?
Thank you!

**Comment by cstritzke** [comment]
Did you try to change your controller port in the topology model to 6644? That might be the problem here. Maybe it makes sense to have another standard port here.

**Comment by ElisaRojas** [comment]
Sorry, but why should I change it to 6644? It makes sense for ODL, because the ODL shim uses that port by default, but not for Ryu :( (I mean, it should not affect at all!). I haven't tried 6644 because I don't want the VM to be created once again from scratch, would that be avoided? If so, please let me know and I can have a try, but to me it does not make sense to change the port...

In fact, I've executed every piece directly on the VM (no via the IDE, but directly on the Vagrant VM created) and the handshake works well, and the core lists components, etc. So there is clearly something different when using the IDE...

![vagrant](https://cloud.githubusercontent.com/assets/2779496/18313039/df6f5d80-750c-11e6-87c5-06e5dc1a98b7.png)

Mininet is still not working though and I'm thinking on how to debug that without a visual interface and no Wireshark :(

Could you please check if this is working for you please? Maybe if you film yourself deploying everything that would be great (in case we are missing something very trivial), but unfortunately things are not working and in this Vagrant VM is quite hard to debug the Engine behavior.
